chainstate/merkle_root.go

```diff
@@ -0,0 +1,77 @@
+package chainstate
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"github.com/BuxOrg/bux/utils"
+)
+
+type merkleRootProvider interface {
+	getName() string
+	verifyMerkleRoots(ctx context.Context, merkleProofs []string) error
+}
+
+// result of single merkle root verification by provider
+type verifyResult struct {
+	isError  bool
+	err      error
+	provider string
+}
+
+// VerifyMerkleRoots will try to verify merkle roots with all available providers
+func (c *Client) VerifyMerkleRoots(ctx context.Context, merkleRoots []string) error {
+	var wg sync.WaitGroup
+	resultsChannel := make(chan verifyResult)
+
+	for _, merkleRootPvdr := range createMerkleRootsProviders(c) {
+		wg.Add(1)
+		go func(provider merkleRootProvider) {
```

Comment:
I don't see a good reason to make this asynchronous, we're actually still waiting for the result in line 42, so the entire function is still blocking (synchronous). 
And from my perspective it's only increasing complexity and gives nothing in exchange.

Reply:
removed